core/src/main/java/org/elasticsearch/common/lucene/search/MultiPhrasePrefixQuery.java

```diff
@@ -150,7 +166,12 @@ public Query rewrite(IndexReader reader) throws IOException {
             }
         }
         if (terms.isEmpty()) {
-            return Queries.newMatchNoDocsQuery("No terms supplied for " + MultiPhrasePrefixQuery.class.getName());
+            // if the terms does not exist we could return a MatchNoDocsQuery but this would break the unified highlighter
+            // which rewrites query with an empty reader.
+            return new BooleanQuery.Builder()
+                .add(query.build(), BooleanClause.Occur.MUST)
+                .add(Queries.newMatchNoDocsQuery("No terms supplied for " + MultiPhrasePrefixQuery.class.getName()),
+                    BooleanClause.Occur.MUST).build();
```

Comment:
Same here, we return a query that returns no result but extractTerms will return the terms that are already expanded.